backtrace-library/src/main/java/backtraceio/library/models/BacktraceData.java

```diff
@@ -110,17 +117,31 @@
     /**
      * Create instance of report data
      *
-     * @param context          current application context
-     * @param report           current report
-     * @param clientAttributes attributes which should be added to BacktraceData object
+     * @param context           current application context
+     * @param report            current report
+     * @param clientAttributes  attributes which should be added to BacktraceData object
      */
     public BacktraceData(Context context, BacktraceReport report, Map<String, Object>
             clientAttributes) {
+        this(context, report, clientAttributes, false);
+    }
+
+    /**
+     * Create instance of report data
+     *
+     * @param context           current application context
+     * @param report            current report
+     * @param clientAttributes  attributes which should be added to BacktraceData object
+     * @param isProguardEnabled is Proguard symbolication enabled?
+     */
+    public BacktraceData(Context context, BacktraceReport report, Map<String, Object>
```

Comment:
Let's think about class design for a second. Is proguard flag necessary to create BacktraceData? I think in most cases - no. I think there is no reason to pass `isProguardEnabled` flag via the constructor. 

`symbolication` property also right now is public. 

Based on this info:
* `symbolication` property should be public? 
* I think we should use properties instead of constructor parameters to avoid unnecessary complexity and also to keep our code clean. Also, it's worth keeping in mind that if we will pass every type of configuration option via backtraceData constructor, our constructor will be huge.

My recommendation:
* set symbolication variable via setter not via the constructor.
* mark `symbolication` variable as internal and set it from `backtraceBase`

Reply:
I marked `symbolication` property as public since all other BacktraceData properties are public. 

Also I don't see `backtraceBase` and `backtraceClient` setting any BacktraceData fields currently, it seems we do pass all the options in from the constructor currently, and I didn't want to break that pattern. @BartoszLitwiniuk also suggested to add `isProguardEnabled` as a constructor parameter in BacktraceData. For now I don't know if we need to make this change to `symbolication`

Reply:
@rqbacktrace I just though it will be better if we will set symbolication field from `backtracebase` or `backtraceClient`. We shouldn't pass backtraceClient/backtracebase instance to backtraceData.

Reply:
As per our conversation I think you are right @konraddysput , I have made this suggested change in the latest commit!